magda-web-client/src/Components/Dataset/Add/DatasetAddCommon.ts

```diff
@@ -180,6 +220,280 @@ type Access = {
     notes?: string;
 };
 
+function dateStringToDate(dateInput: any): Date | null {
+    if (dateInput instanceof Date && !isNaN(dateInput.getTime())) {
+        return dateInput;
+    }
+    if (!dateInput || typeof dateInput !== "string") {
+        return null;
+    }
+    const d = new Date(dateInput);
+    if (isNaN(d.getTime())) {
+        return null;
+    }
+    return d;
+}
+
+function populateDcatDatasetStringAspect(data: RawDataset, state: State) {
+    const datasetDcatString = data?.aspects["dcat-dataset-strings"];
+    if (!datasetDcatString) {
+        return;
+    }
+    state.dataset = {
+        ...state.dataset,
+        title: datasetDcatString.title,
+        description: datasetDcatString.description
+    };
+
+    if (typeof datasetDcatString?.languages === "string") {
+        state.dataset.languages = [datasetDcatString.languages];
+    } else if (datasetDcatString?.languages?.length) {
+        state.dataset.languages = datasetDcatString.languages;
+    }
+
+    if (datasetDcatString?.accrualPeriodicity) {
```

Comment:
Yeah~ you are correct. It's a bit redundant.
```
state.dataset.accrualPeriodicity = datasetDcatString?.accrualPeriodicity
```
will not make program break and will assign `undefined` to state.dataset.accrualPeriodicity .
I just try to avoid to set a key with `undefined` value on the object `dataset` so that even other places, where we haven't used optional chaining yet, have less chance to break.
What's your idea?

Reply:
:man_shrugging: yeah that's a fair point, all good :)